src/web/services/projects/view.rs

```diff
@@ -24,10 +34,21 @@ pub async fn project(
 
     let project = project.unwrap();
 
+    let can_edit = authorization.can_edit();
+
+    if !authorization.can_view() {
+        return Err(TelescopeError::BadRequest {
```

Comment:
What about returning a permission denied error instead of a generic bad request one?